File: solutions/0027-remove-element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int last = nums.length -1;
        int i=0;
        while(i<=last){
            if(nums[i]==val){
                nums[i] = nums[last];
                last--;
            }
            else{
                i++;
            }
        }
        return last+1;
}
    
}

File: solutions/0027-remove-element/solution.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i=0;
        int curr=0;
        int count = 0;
        while(curr<nums.size()){
            if(nums[curr]==val){
                curr++;
            }
            else{
                nums[i]=nums[curr];
                i++;
                curr++;
                count++;
            }
        }
        return count;
    }
};
